pkg/machines/components/nicAdd.jsx

```diff
@@ -0,0 +1,255 @@
+                        type="radio"
+                        name="set-mac"
+                        checked={dialogValues.setNetworkMac}
+                        onChange={e => onValueChanged('setNetworkMac', true)}
+                        className={dialogValues.setNetworkMac ? "active" : ''} />
+                    {_("Set manually")}
+                </label>
+                <input id={`${idPrefix}-mac`}
+                    className='form-control'
+                    type='text'
+                    disabled={!dialogValues.setNetworkMac}
+                    value={dialogValues.networkMac}
+                    onChange={e => onValueChanged('networkMac', e.target.value)} />
+            </div>
+        </>
+    );
+};
+
+const PermanentChange = ({ idPrefix, onValueChanged, dialogValues, provider, vm }) => {
+    // By default for a running VM, the iface is attached until shut down only. Enable permanent change of the domain.xml
+    if (!provider.isRunning(vm.state))
+        return null;
+
+    return (
+        <>
+            <label className="control-label"> {_("Persistence")} </label>
+            <label className='checkbox-inline'>
+                <input id={`${idPrefix}-permanent`}
+                       type="checkbox"
+                       checked={dialogValues.permanent}
+                       onChange={e => onValueChanged('permanent', e.target.checked)} />
+                {_("Always attach")}
+            </label>
+        </>
+    );
+};
+
+export class AddNICAction extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.state = this.initialState;
+        this.open = this.open.bind(this);
+        this.close = this.close.bind(this);
+        this.add = this.add.bind(this);
+        this.onValueChanged = this.onValueChanged.bind(this);
+        this.dialogErrorSet = this.dialogErrorSet.bind(this);
+    }
+
+    get initialState() {
+        const state = {
+            showModal: false,
+            dialogError: undefined,
+            networkType: "network",
+            networkSource: undefined,
+            networkModel: "virtio",
+            setNetworkMac: false,
+            networkMac: "",
+            permanent: false,
+            addDisabled: false,
+        };
+
+        const availableSources = this.props.networks.map(network => network.name);
+        if (availableSources.length > 0)
+            state.networkSource = availableSources[0];
+
+        return state;
+    }
+
+    onValueChanged(key, value) {
+        const stateDelta = { [key]: value };
+
+        this.setState(stateDelta);
+
+        if (key == 'networkType') {
+            let addDisabled = false;
+
+            if (value == 'network' || value == 'direct') {
+                let availableSources;
+                if (value == 'network')
+                    availableSources = this.props.networks.map(network => network.name);
+                else if (['direct'].includes(value))
```

Comment:
Please use consistent way to compare with the string. 
Like if (value == 'network') use
else if (value == 'direct')